### PR TITLE
feat(mocks): generate fakes and mocks in a container

### PR DESCRIPTION
**Pull request by njhale**
### Description

Adds the `container-mockgen` make rule for generating fakes and mocks in a container.

### CODE REVIEW

File: mockgen.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM golang:1.10
```
Comment (jpeeler) [inline mockgen.Dockerfile]: Does this not work correctly with 1.11?

Comment (njhale) [reply mockgen.Dockerfile]: doesn't seem to for me.

File: Makefile [hunk 2]
```diff
@@ -117,26 +118,33 @@ codegen:
 	$(CODEGEN) all $(PKG)/pkg/package-server/client $(PKG)/pkg/package-server/apis "packagemanifest:v1alpha1"
 
 container-codegen:
-	docker build -t olm:codegen -f codegen.Dockerfile .
+	docker build -t olm:codegen -f mockgen.Dockerfile .
```
Comment (ecordell) [inline Makefile]: shouldn't this still be `codegen.Dockerfile`?

Comment (njhale) [reply Makefile]: yup

**Pull request merged**